File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/dominio/calculo/relatorio/IrpfHonorariosAtualizacaoJRAdapter.java
/*
 * Decompiled with CFR 0.152.
 * 
 * Could not load the following classes:
 *  net.sf.jasperreports.engine.data.JRBeanCollectionDataSource
 */
package br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio;

import br.jus.trt8.pjecalc.base.comum.relatorio.JRAdapter;
import java.util.Date;
import net.sf.jasperreports.engine.data.JRBeanCollectionDataSource;

public abstract class IrpfHonorariosAtualizacaoJRAdapter
extends JRAdapter {
    public abstract JRBeanCollectionDataSource getIrpfHonorariosPorEventos();

    public abstract JRBeanCollectionDataSource getIrpfHonorariosDoSaldo();

    public abstract Date getDataLiquidacaoAtualizacao();

    public abstract class IrpfHonorariosPorEventoJRAdapter
    extends JRAdapter {
        public abstract Date getDataEvento();

        public abstract JRBeanCollectionDataSource getOcorrencias();
    }
}

File: src/lib/pjecalc/core/dominio/calculo/relatorio/irpf-honorarios-atualizacao-jr-adapter.ts
/**
 * PJe-Calc v2.15.1 — IrpfHonorariosAtualizacaoJRAdapter (abstract)
 * Porte TS-adaptado de:
 *   br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio.IrpfHonorariosAtualizacaoJRAdapter
 *
 * Ref Java: pjecalc-fonte/.../relatorio/IrpfHonorariosAtualizacaoJRAdapter.java
 *
 * Relatório de IRPF aplicado sobre honorários em atualização (faixas PF/PJ).
 */
import type Decimal from 'decimal.js';
import { JRAdapter, JRAdapterDataSource } from './jr-adapter';

export abstract class IrpfHonorariosAtualizacaoJRAdapter extends JRAdapter {
  abstract getHonorariosAtualizados(): JRAdapterDataSource<IrpfHonorariosAtualizacaoItemAdapter>;
  abstract getTotalBaseImposto(): Decimal;
  abstract getTotalValorImpostoRenda(): Decimal;
}

export abstract class IrpfHonorariosAtualizacaoItemAdapter extends JRAdapter {
  abstract getDataEvento(): Date;
  abstract getDescricao(): string;
  abstract getTipoImpostoRenda(): string; // PF / PJ
  abstract getBaseImposto(): Decimal;
  abstract getValorInicialFaixa(): Decimal;
  abstract getValorFinalFaixa(): Decimal;
  abstract getValorAliquota(): Decimal;
  abstract getValorDeducao(): Decimal;
  abstract getValorImpostoRenda(): Decimal;
}
